core/src/main/java/org/commonjava/indy/core/ctl/ContentController.java

```diff
@@ -157,6 +152,16 @@ public String getContentType( final String path )
         return mimeTyper.getContentType( path );
     }
 
+    public String getNPMContentType( final String path )
```

Comment:
Sorry, but this can't be in something generic like the ContentController. Can we add it to the NPMContentAccessHandler instead? We're trying to take the package-specific details out of the core Indy functionality, and this adds NPM-specific handling back in.

If multiple NPM-layer components need to call a method like this, I guess we might need a new controller-layer component to allow them to share logic. Or maybe a utility class. But if we don't need to call from multiple components, I'd just put it in NPMContentAccessHandler.

Reply:
@jdcasey Ok, I'll move this to NPMContentAccessHandler since now it's only used for the handler's headers handle.
The reason I move doHead method to npm module is that I'm aware npm needs some specific logic to allow it to pass the client build requirements (especially npm has path map solution which will direct it to be different with maven), maybe we need some npm header enhancements to do in not so far away (I'm not sure, but I think this is necessary).
And for now, it has two diffs from ContentAccessHandler doHead:
1. http-meta fetching, npm will fetch it as the mapping path like: **/package.json.http-metadata.json (original will point to **.http-metadata.json, that will result wrong headers setting like content-lengh which is a directory not json file exactly)
2. setInfoHeaders param, npm has it own contentType logic to pass

Reply:
:+1: